src/App.js

```diff
@@ -1,16 +1,64 @@
 import React from 'react';
 import './App.css';
+import {useState} from 'react';
+import ChatLog from './components/ChatLog.js';
+import ColorChoice from './components/ColorChoice.js';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const[chatData, setChatData] = useState(chatMessages);
+
+  const calcTotalLikes = (chatData) => {
+    return chatData.reduce((total, entry) => {
+      let likeCount = entry.liked ? 1: 0;
+      return total + likeCount;
+  }, 0)
+
+  
+
+};
+const totalLikesTally =  `${calcTotalLikes(chatData)} ❤️s`
+
+const setColorCallback = (color) => {
+  
+  return color;
```

Comment:
For this event handler, I think you need to have logic that sets the color value, not return it. 
```
const setColorCallback = (sender, color) => {
    setChatData((messageData) =>
      messageData.map((message) => {
        if (message.sender === sender) {
          return {...message, color: color };
        } else {
          return message;
        }
      })
    );
```